mh/time_utils.py

```diff
@@ -20,9 +20,44 @@
 """
 
 import datetime
+from math import floor
+
 
 EPOCH = datetime.datetime(1970, 1, 1)
+TICKS_PER_CYCLE = 3000
 
 
 def datetime_to_int(dt):
     return int((dt - EPOCH).total_seconds())
+
+JHEN_EVENT_OFFSET = 14  # Cycle of 14 days
+FOG_START = 0  
+JHEN_START = 1  # 1 day of fog
+JHEN_END = 3  # Followed by 2 days of sandstorm
+
+
+def current_tick():
+    return 4294967295 - datetime_to_int(datetime.datetime.now())
+
+
+def current_server_time():
+    return datetime_to_int(datetime.datetime.now())
+
+
+def get_jhen_cycle_time(time=current_server_time()):
+    current_server_day = int(floor(time/(24*60*60)))
+    if (current_server_day % JHEN_EVENT_OFFSET) >= JHEN_END:
+        next_day = current_server_day + (JHEN_EVENT_OFFSET - (current_server_day % JHEN_EVENT_OFFSET))
+    else:
+        next_day = current_server_day - (current_server_day % JHEN_EVENT_OFFSET)
+    
+    return next_day*24*60*60
+
+def get_fog_start(time=current_server_time()):
+    return int(get_jhen_cycle_time(time) + FOG_START*24*60*60)
+
+def get_sandstorm_start(time=current_server_time()):
+    return int(get_jhen_cycle_time(time) + JHEN_START*24*60*60)
+
+def get_sandstorm_end(time=current_server_time()):
+    return int(get_jhen_cycle_time(time) + JHEN_END*24*60*60)
```

Comment:
- Can't these be a simple function returning a tuple instead? You're calling the same function multiple times.
 - `time` as a variable name should be avoided as it's shadowing the `time` module.
 - `24 * 60 * 60` should be stored to a variable or a constant with a meaningful name.
 - IIRC, you don't need to import `floor` in your case and can use the `//` operator: `time // (24*60*60)`
 - Can't a modulo be used instead of the `if (...) >= JHEN_END: (...) else:`?

Reply:
* Turned into a single function returning a tuple
* No longer shadowing `time` module
* Stored `24 * 60 * 60` to a constant
* Removed `floor` import and switched to `//`
* I'm not sure what you mean by this one, I'm already using a modulus in the `if` statement to determine if the day in the event cycle is passed the end of the jhen event